subt/docker/base/Dockerfile

```diff
@@ -84,7 +84,9 @@ WORKDIR /osgar-ws
 
 RUN /usr/bin/python3 -m venv env && \
   /osgar-ws/env/bin/pip install --no-cache-dir \
+  "http://osgar.robotika.cz/subt/pip/tensorflow/tensorflow-2.2.0rc3-cp36-cp36m-linux_x86_64.whl" \
   "msgpack==1.0.0" \
   "numpy==1.18.2" \
   "opencv-python==4.2.0.32" \
-  "pyzmq==19.0.0"
+  "pyzmq==19.0.0" \
+  "torch==1.4.0" \
```

Comment:
Would you mind switching to 1.5.0? It was published after my initial experiments and contains some important improvements, namely a fix to make 1x1 convolutions work, which allows faster architectures, such as MobileNet.

Reply:
will do

Reply:
I tried that locally and tensorflow test has failed when pytorch got updated to 1.5.0. I will investigate.

Reply:
```
2020-05-04 14:18:05.532616: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1
2020-05-04 14:18:05.532721: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-05-04 14:18:05.533093: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce GTX 1050 computeCapability: 6.1
coreClock: 1.493GHz coreCount: 5 deviceMemorySize: 3.95GiB deviceMemoryBandwidth: 104.43GiB/s
2020-05-04 14:18:05.533228: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/ros/melodic/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
2020-05-04 14:18:05.534360: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-05-04 14:18:05.535417: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10
2020-05-04 14:18:05.535598: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10
2020-05-04 14:18:05.536795: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10
2020-05-04 14:18:05.537541: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10
2020-05-04 14:18:05.540088: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2020-05-04 14:18:05.540110: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1598] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
```

Reply:
<pre>root@larsa:/osgar-ws# find / -name &apos;libcudart.so.10.*&apos;
/usr/local/cuda-10.2/targets/x86_64-linux/lib/libcudart.so.10.2
/usr/local/cuda-10.2/targets/x86_64-linux/lib/libcudart.so.10.2.89
</pre>

Reply:
Sorry, I don't know what I was thinking. The cuda base image is 10.2 instead of 10.1. I am also wondering why the previous tests went ok :-/

Reply:
Ok, I probably know what happened - the older pytorch might have brought cuda 10.1 along for tensorflow to use. The new pytorch is probably cuda 10.2.

I am looking at https://pytorch.org/get-started/locally/ and it seems to suggest that for getting 1.5.0 with cuda 10.1 we should use this command

```
pip install torch==1.5.0+cu101 torchvision==0.6.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html
```

Can you confirm? Do we need `torchvision`?

Reply:
I use torchvision during training and don't use it during inference. So it is a "nice to have", but not necessarily a "must have".